Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/HolographicCamera/HolographicCameraOriginAnchor.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Extensions.Experimental.Socketer;
+using Microsoft.MixedReality.Toolkit.Extensions.Experimental.SpectatorView.WorldAnchors;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using UnityEngine;
+using UnityEngine.XR.WSA;
+
+namespace Microsoft.MixedReality.Toolkit.Extensions.Experimental.SpectatorView.HolographicCamera
+{
+    [RequireComponent(typeof(WorldAnchorManager))]
+    [RequireComponent(typeof(TCPConnectionManager))]
+    public class HolographicCameraOriginAnchor : MonoBehaviour
+    {
+        private const string AnchorName = "WorldRoot";
+        private WorldAnchorManager worldAnchorManager;
+        private TCPConnectionManager tcpConnectionManager;
+
+        public bool IsAnchorLocated
+        {
+            get
+            {
+                WorldAnchor anchor = GetComponent<WorldAnchor>();
+                return anchor != null && anchor.isLocated;
+            }
+        }
+
+        private void Start()
+        {
+            worldAnchorManager = GetComponent<WorldAnchorManager>();
+            worldAnchorManager.AttachAnchor(gameObject, AnchorName);
+
+            tcpConnectionManager = GetComponent<TCPConnectionManager>();
+            tcpConnectionManager.OnReceive += TcpConnectionManager_OnReceive;
+        }
+
+        private void TcpConnectionManager_OnReceive(IncomingMessage data)
+        {
+            using (MemoryStream stream = new MemoryStream(data.Data))
+            using (BinaryReader reader = new BinaryReader(stream))
+            {
+                string command = reader.ReadString();
+
+                switch (command)
+                {
+                    case "CreateSharedAnchor":
+                        {
+                        }
+                        break;
```

Comment:
Should this do anything? or is this just a first pass for future functionality?

Reply:
I don't like the design of this.

There should be a protocol that handles sorting of commands in a more standardized way.

Personally, I'm a fan of Google's protobuf protocol, as the protocol templates it produces are compiled natively in all languages it supports, including c++ and c#, but also java.

This means you can parse it in a native app on any platform supported by Azure spatial anchors.

Reply:
This is just a first pass for future functionality, the next PR I have pending has this block filled in.

RE: protobof. This is definitely worth looking into. This PR is just the tip of porting a large amount of currently-working code from an internal repository, but I'll investigate what work would be required to change the networking solution to something like protobuf. We also have some tasks to improve performance of the state synchronization, and this could tie into that work as well.